exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -35,6 +42,8 @@ import (
 	"sigs.k8s.io/cluster-api/util/predicates"
 )
 
+const caBundleInjectFromSecretAnnotation = "runtime.cluster.x-k8s.io/inject-ca-from-secret" //nolint:gosec // this is not a credential.
```

Comment:
nit: let's move this into the api package to ExtensionConfig (maybe in a common_types.go like in the clusterv1 api package)

Reply:
I've added it to the `extensionconfig_types.go`. There are already some consts for Discovery and the annotation is specific to the `ExtensionConfig`, and thus not a common annotation.

Reply:
Sounds good!